pkg/nmi/managed.go

```diff
@@ -87,7 +89,20 @@ func (mc *ManagedClient) GetToken(ctx context.Context, rqClientID, rqResource st
 			clientSecret = string(v)
 			break
 		}
-		token, err := auth.GetServicePrincipalToken(tenantid, clientID, clientSecret, rqResource)
+		token, err := auth.GetServicePrincipalToken(adEndpoint, tenantID, clientID, clientSecret, rqResource)
+		return token, err
+	case aadpodid.ServicePrincipalCertificate:
+		tenantID := azureID.Spec.TenantID
+		adEndpoint := azureID.Spec.ADEndpoint
+		klog.Infof("matched identityType:%v adendpoint:%s tenantid:%s clientid:%s resource:%s",
```

Comment:
The error is here - 
```
Infof format %s has arg idType of wrong type github.com/Azure/aad-pod-identity/pkg/apis/aadpodidentity.IdentityType
```

Please change the order of args - "idType, adEndpoint tenantID, utils.RedactClientID(clientID), rqResource"

Reply:
Got it, thanks Anish